lib/linter-views.coffee

```diff
@@ -1,183 +1,64 @@
-BottomTab = require './views/bottom-tab'
-BottomStatus = require './views/bottom-status'
-Message = require './views/message'
 
-class LinterViews
-  constructor: (@state, @linter) ->
-    @showPanel = true
-    @showBubble = true
-    @underlineIssues = true
-
-    @messages = new Set
-    @lineMessages = new Set
-    @markers = []
-    @statusTiles = []
-    @tabPriority = ['File', 'Project', 'Line']
+{CompositeDisposable} = require('atom')
 
-    @tabs =
-      Line: new BottomTab()
-      File: new BottomTab()
-      Project: new BottomTab()
+BottomPanel = require('./views/bottom-panel')
+BottomContainer = require('./views/bottom-container')
+BottomStatus = require('./views/bottom-status')
+Message = require('./views/message')
 
-    @panel = document.createElement 'div'
+class LinterViews
+  constructor: (@linter) ->
+    @state = @linter.state
```

Comment:
There is a reason why I used scaffolding for states in https://github.com/AtomLinter/Linter/pull/669. If you ever decide to store something else in state, it will kind of blow up in your face (first, it won't be compatible between versions, which can lead to a whole lot of unexpected problems for users, and second, it's better to recreate it on serialization, in case something goes wrong and state you received would be corrupted in some way).